lms/services/auto_grading.py

```diff
@@ -1,8 +1,54 @@
+from sqlalchemy import select
+
 from lms.js_config_types import AnnotationMetrics
-from lms.models import AutoGradingConfig
+from lms.models import AutoGradingConfig, GradingSync, GradingSyncGrade, LMSUser
 
 
 class AutoGradingService:
+    def __init__(self, db):
+        self._db = db
+
+    def get_in_progress_sync(self, assignment) -> GradingSync | None:
+        return self._db.scalars(
+            self.search_query(
+                assignment=assignment, statuses=["scheduled", "in_progress"]
+            )
+        ).one_or_none()
```

Comment:
I guess it should not be possible to end up with more than one sync scheduled/in_progress for the same assignment, right?

Reply:
Yes, I created an issue to include a CHECK constraints at the DB level if we don't find any other issue with this design.